numpy/lib/ufunclike.py

```diff
@@ -43,15 +103,11 @@ def fix(x, y=None):
     array([ 2.,  2., -2., -2.])
 
     """
-    x = nx.asanyarray(x)
-    y1 = nx.floor(x)
-    y2 = nx.ceil(x)
-    if y is None:
-        y = nx.asanyarray(y1)
-    y[...] = nx.where(x >= 0, y1, y2)
-    return y
-
-def isposinf(x, y=None):
+    return _where(nx.greater_equal(x, 0), nx.floor(x), nx.ceil(x), out)
```

Comment:
The version of `where` you have above is not subclass-safe, at least not for `Quantity` (where, in general, the two arrays might have different units). For this particular function, it might nevertheless work, but I would suggest not introducing a new function. Instead, perhaps the following (I checked it passes all tests):
```
    out = nx.asanyarray(nx.floor(x, out=out))
    out = nx.ceil(x, out=out, where=(x < 0))
    return out if out.shape else out[()]
```